### PR TITLE
Phone header menu, Profile folded into Settings, editable account

### PR DESCRIPTION
**Pull request by jlamoreaux**
## Summary

Two commits from the profile/settings UX review.

**1. The header fits a phone.** Signed in, the nav's five links were one unwrappable row, so below about 460px the page scrolled sideways and the wordmark and "logout" were clipped off either edge. Below 640px the links now collapse behind a "menu" button beside the wordmark and open as a full-width list beneath it. The toggle is a checkbox driven purely by CSS (no client script), visually hidden rather than `display:none` so it keeps its place in the tab order and the label shows its focus ring. The collapse keys off the toggle's presence, so the docs site's header (which shares `nav-css.ts`) and the signed-out header keep their inline links.

**2. Profile folded into Settings, and the account is editable.** The two pages opened with the same Account card and each pointed at the other for the rest. Settings now holds everything with jump links under the heading; `/profile` redirects to `/settings#account`. The header drops the *profile* link, marks the link for the current page (`aria-current`), and shows the display name when one is set.

The Account section gains the two things about an account that can change:
- a free-form **display name** (any time; shown in the header instead of the username), and
- the **username**, only while the account owns no projects. The username is the namespace every project URL, clone URL, KV key and backing repository is keyed under, and nothing rewrites those, so while any project exists the form is replaced by a sentence saying why. The check fails closed if the claims table or the project listing cannot be read. The signup forms and getting-started guide now describe the username the same way.

Migration `045_user_display_name.sql` adds the column; `046_namespace_claims.sql` adds the claims table described below.

**Settings findings fixed alongside:** the API token, connected-app and agent tables were outside the scrolling wrapper (page was 741px wide at a 390px viewport); the invite share link broke at any character into 300px-tall rows; an email overflowed the Account list at 320px; the global reset had removed all spacing between help copy, list, checkbox and forms; help paragraphs of 50–90 words are now a sentence or two (analytics detail behind a `<details>`); native select and checkbox styled; form controls, table buttons and phone menu rows meet a 40px minimum; card headings are `h2` under the page `h1`; dates read "Sep 2, 2026" instead of the server locale's "9/2/2026"; Rotate/Disable on the legacy key are plain buttons so red stays in the danger zone.

**Rename vs. project creation.** Project records live in KV, whose listings are only eventually consistent, so a rename could never learn from KV alone whether a project had just been created. D1 orders the two instead: every personal project-creation path (`POST /api/projects`, the single-import route, bulk import) inserts a row into `namespace_claims` before it writes the project's KV entry, and `renameUser` refuses inside its own `UPDATE` (`NOT EXISTS` over the claims for that owner). Whichever reaches D1 first wins, and the loser sees it: a rename after a claim gets a 403; a claim after a rename is caught by `confirmOwnerNamespace`, which re-reads the owner after the KV write and withdraws the project (KV entry, backing repository, claim), failing closed if the owner cannot be re-read twice.

A claim's release is best effort, so a claim is not trusted forever: past `CLAIM_GRACE_MS` (15 minutes, well beyond KV's settle time) the rename check looks each claim's project up in KV and drops a claim with no project behind it, whether its release failed or the project was deleted later. Within the window a claim is taken at its word, which is what the table is for. The KV listing is still consulted for projects that predate the table, and the rename still re-lists the old namespace afterwards and reverts if one appears.

## Related issues

None closed. Touches part of the design audit in #310 (active nav state, header layout) and #309 (red buttons outside the danger zone); the rest of both issues is untouched and they stay open.

## Type of change

- [x] Bug fix
- [x] New feature
- [ ] Refactor / cleanup
- [ ] Documentation
- [ ] Other:

## How was this verified?

- [x] `npm run typecheck`
- [x] `npm test`
- [x] `npm run lint`
- [x] `website`: `mirror-header --check` and `mirror-docs --check` in sync
- Rendered the header and Settings at 320, 390, 600, 640, 641, 768 and 1280px in headless Chromium: no horizontal overflow at any width, menu toggles by mouse and by Tab + Space, heading order `h1 → h2`, no visible control under 32px.
- Smoke-checked the preview environment over HTTP: health, the served stylesheet, the `/profile` redirect, the signup hint, and the anonymous `/settings` redirect.
- The claim helpers, the rename's `NOT EXISTS` refusal, claim release on withdrawal, and stale-claim reconciliation run against the real schema (`tests/helpers/sqlite-d1.ts`), not a string-matching stub.

## Checklist

- [x] Tests added or updated for the change (coverage thresholds still pass)
- [x] `CHANGELOG.md` updated (`## [Unreleased]`) if this is user-visible
- [x] Public docs updated if this changes user-facing config, API shape, or evaluator/policy behavior — `docs/user-guide/getting-started.md` and `faq.md`; mirrors regenerated
- [x] No secrets, tokens, or personal data committed
- [x] The web UI change (if any) stays server-rendered with no client-side JavaScript

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01UoLc6AEsrBbvGVUVEANSwh

## Summary by CodeRabbit

- **New Features**
  - Added account settings for editing display names and eligible usernames.
  - Added invite-code access within **Settings → Account**.
  - Added responsive mobile navigation and clearer active-page indicators.
  - Improved responsive tables, forms, and settings layouts.
- **Bug Fixes**
  - Strengthened project namespace protection during creation and imports.
  - `/profile` now redirects to Settings.
  - Improved mobile navigation, invite links, account emails, and table usability.
- **Documentation**
  - Updated account, username, and invite-code guidance, including username restrictions after creating projects.

### CODE REVIEW

**Comment by jlamoreaux** [comment]
On the two PR-Agent findings, both addressed in 5b36712:

**Misplaced comment** — `renderSettings`'s doc comment is back on `renderSettings`.

**Rename race** — narrowed rather than closed, and the residual is now stated in the code. After the `UPDATE`, the route lists the old namespace again: anything there was created between the first listing and the rename, so the name is put back and the user is asked to retry (409). A project created after that point reads the new username. What remains is a creation that read the old name before the `UPDATE` and wrote KV after the second listing (a window the width of that one request), plus KV listings being only eventually consistent across edges. Neither can be closed from this route: the two stores have no shared transaction, and KV offers no strong read. Serializing per account would need a lock in D1 that every project-creation path also takes, which is a larger change than this PR should carry; happy to open a follow-up if that's wanted.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by jlamoreaux** [comment]
All four CodeRabbit findings are addressed in 52bc13a, and it corrects my earlier comment on the race.

**Rename vs. project creation.** Rather than a per-user lock across every creation path, both sides now check after their own write. The rename re-lists the old namespace and reverts if a project appeared (5b36712). Each personal creation path (`POST /api/projects`, the single import, bulk import) now re-reads the owner after writing KV and withdraws the entry and its backing repository if the username moved (`confirmOwnerNamespace`). A project written under the old name after the rename's second listing is seen by the creation's re-read, and one written before it is seen by the listing, so no interleaving leaves a project under a namespace nobody owns. D1 reads are strongly consistent; the residual is KV's eventual consistency across edges on the listing, which no application-level check can close.

**Found while doing this:** the ownership check listed projects under the bare username, but every project key carries the `@` prefix, so it found nothing and would have allowed a rename for an account that owns projects. My test mocked the listing and missed it. The listing now goes through `getUserNamespace` (moved from two private copies in the project routes to `src/types.ts`), and the test asserts the `@`-prefixed argument.

**Also:** `\p{Cf}` added to the display-name normaliser with tests for bidi overrides and zero-width characters; `clip-path: inset(50%)` replaces `clip` in `nav-css.ts` and the mirrored `header.css`; the layout comments now state the constraint rather than the code.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by jlamoreaux** [comment]
@coderabbit review

**Comment by jlamoreaux** [comment]
@coderabbitai review

Re-requesting now that the hourly limit has passed: 534de02 addresses the three findings from the last pass (guard fails closed on an unreadable owner or a failed KV delete; the menu checkbox is out of the desktop tab order; the display name reaches every page's header). CI is green on that head and all threads are resolved.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/storage/project-namespace.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+      409,
+      { namespace, slug },
+    ),
+  );
+}
+
+/** The owner row, with one retry: a single failed read must not decide a project's fate. */
+async function readOwner(db: D1Database, ownerId: string, logger: Logger): Promise<User | null> {
+  for (let attempt = 0; attempt < 2; attempt++) {
+    const result = await getUser(db, ownerId, logger);
```
[the remark this answers is not in the document]
Comment (jlamoreaux) [reply src/storage/project-namespace.ts]: Addressed in b376db6: each attempt in `readOwner` is wrapped in try/catch, a thrown read counts like a failed one, and two of them return `null` so `confirmOwnerNamespace` withdraws. Tests in `tests/project-namespace-guard.test.ts` reject both reads and assert withdrawal, and reject the first read only and assert the project is kept.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by jlamoreaux** [comment]
b376db6 closes the rename/creation ordering gap CodeRabbit raised in the last round, this time in D1 rather than with more KV listings.

**What changed.** Migration `046_namespace_claims.sql` adds a small table keyed by `(namespace, slug)` with an `owner_id`. Each personal project-creation path (`POST /api/projects`, the single-import route, bulk import) inserts a claim right before it writes the project's KV entry and releases it if that write fails. `renameUser` now refuses inside its own statement: `UPDATE users SET username = ? WHERE id = ? AND NOT EXISTS (SELECT 1 FROM namespace_claims WHERE owner_id = ?)`. So the two operations are ordered by D1, which is strongly consistent, and each side sees the other: a rename that loses gets a 403 from the UPDATE; a creation that loses is caught by `confirmOwnerNamespace`, which re-reads the owner after the KV write and withdraws the entry, the backing repository and the claim. `readOwner` also catches a rejected read now, so a thrown D1 error reaches that withdrawal path.

**Design choice worth a look.** Claims outlive their projects: the username is fixed once the first project has ever been created, deleted or not. That matches the wording already in the docs and the settings page ("until you create your first project"), and it means an old `@name/...` URL never resolves to a different account later. If you would rather deleting every project re-opens the rename, `deleteProject` would release the claim and the wording would change; I went with the safer rule.

The KV listing is still consulted before a rename for projects that predate the table, and the post-rename re-listing stays as a backstop for those. The table is in the backup allow-list.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/storage/project-namespace.ts [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+  } catch (error) {
+    logger.warn("Could not delete the backing repository of a withdrawn project", {
+      namespace,
+      slug,
+      error: error instanceof Error ? error.message : String(error),
+    });
+  }
+  // A claim left behind would fix a username for a project that never
+  // existed; the release is best effort, and a leftover only errs on the safe side.
+  await releaseNamespaceClaim(env.DB, namespace, slug, logger);
```
[the remark this answers is not in the document]
Comment (jlamoreaux) [reply src/storage/project-namespace.ts]: Addressed in 030f5fd, by reconciliation rather than a retry queue. A claim is no longer trusted past a grace window (`CLAIM_GRACE_MS`, 15 minutes, well beyond KV's settle time): `ownerHasClaims` now reads the owner's claims newest first, counts a young one as it stands, and for an older one looks the project up in KV. A claim with no project behind it is dropped on the spot, matching on `created_at` so a claim refreshed by a re-creation of the same slug survives. That covers the shared withdrawal path and all three creation failure paths without any of them having to succeed at the release, and it also covers a project deleted later. It fails closed: a KV read that errors or throws, or a drop that fails, is reported as an error rather than counted either way.

This also settles the user-facing rule on the side the PR originally stated: the username can change while the account owns no projects, and deleting them all frees it again after the window. Copy, docs, migration comment and the earlier PR comment's "claims outlive projects" choice are updated accordingly.

Tests in `tests/project-namespace-guard.test.ts` ("stale namespace claims"): fresh claim counted without a KV read, old claim with a live project counted, stale claim dropped and not re-read, a later live claim still counted past a stale one, KV error and KV throw both fail closed, a failed drop reported as an error, and a refreshed claim left alone.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by jlamoreaux** [comment]
@coderabbitai review

---
_Generated by [Claude Code](https://claude.ai/code)_

File: docs/user-guide/getting-started.md [hunk 3]
```diff
@@ -27,9 +27,11 @@ Sign in with any of:
 
 All three resolve to the same email-identity account, so you can mix methods.
 Every method asks you to choose a **username** the first time: it is your
-namespace (`@you/project` in every URL and clone URL) and cannot be changed
-later. GitHub and Google sign-in suggest one from your handle or email address
-and let you pick another before the account exists.
+namespace (`@you/project` in every URL and clone URL). You can change it in
+Settings while you own no projects; while any exists it is fixed, because
```
[the remark this answers is not in the document]
Comment (jlamoreaux) [reply docs/user-guide/getting-started.md]: Addressed in e07c814: the getting-started guide (and its website mirror) now says that after deleting your last project the form can take up to 15 minutes to reappear, and the CHANGELOG entry says a claim is trusted for 15 minutes with the same consequence.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/ui/pages/settings.tsx [hunk 4]
```diff
@@ -158,8 +192,97 @@ const ApiTokenRow: FC<{ token: ApiTokenSummary; now: number }> = ({ token, now }
...
+            maxlength={39}
+            pattern={USERNAME_PATTERN}
+            title={USERNAME_RULES}
+            autocomplete="username"
+            spellcheck={false}
+          />
+        </label>
+        <p class="settings-help">
+          Your namespace: <code>@{user.username}/…</code> in every project and clone URL. It can be
+          changed only while you own no projects.
```
[the remark this answers is not in the document]
Comment (jlamoreaux) [reply src/ui/pages/settings.tsx]: Addressed in e07c814. The help text under the username field now reads "It can be changed only while you own no projects; after deleting your last project, allow up to 15 minutes", the sentence shown in place of the form says the same, and the 403 from the claims check (route and `renameUser`) carries it too. The assertions in `tests/account-settings.test.tsx` cover the surviving phrases.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: tests/user-profile-storage.test.ts [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+  });
+});
+
+/**
+ * The rename's refusal lives inside its UPDATE (`NOT EXISTS` over
+ * namespace_claims), so it needs the real schema: a stub cannot tell whether
+ * the subquery is right.
+ */
+describe("renameUser against namespace claims", () => {
+  /** One user, alice, in the real schema. */
```
[the remark this answers is not in the document]
Comment (jlamoreaux) [reply tests/user-profile-storage.test.ts]: Removed in e07c814.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by jlamoreaux** [comment]
e07c814 addresses the outside-diff finding from the last CodeRabbit round (a withdrawal whose KV delete fails twice left the project reachable with nothing tracking it), plus the three inline notes.

**Durable withdrawal.** `withdraw` still tries the KV delete twice. When both fail, `scheduleWithdrawal` now hands the project to the existing deletion pipeline: `captureDeletionTarget` + `createDeletionJob(kind: "project")`, the same cascade a user-requested delete goes through, re-driven by the five-minute sweep with its attempt budget and operator trail. The inventory is captured the normal way when it can be; since the project is seconds old, when the capture itself fails (KV is what just failed) a target built from the entry alone is complete except for the collision scan, which is assumed positive so the cascade never touches rows keyed by a bare name another tenant might share. The request still returns a 500, now saying the removal is scheduled. Only when the job cannot be recorded either does it fall back to the error-level log and "contact support" message, since that is the one outcome that needs a person. `confirmOwnerNamespace` takes the `ProjectEntry` instead of namespace and slug so the job target can be built without a KV read; all three creation paths pass the entry they just wrote.

Tests in `tests/project-namespace-guard.test.ts`: a failed entry delete schedules a job with the captured inventory and keeps the repository and claim for it; a failed capture builds the target from the entry with `nameCollision: true`; a failed job creation reports the operator case.

Full suite, typecheck, lint and the doc mirrors are clean locally.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by jlamoreaux** [comment]
@coderabbitai review

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by jlamoreaux** [comment]
@coderabbitai review

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/ui/pages/settings.tsx [hunk 6]
```diff
@@ -158,8 +192,98 @@ const ApiTokenRow: FC<{ token: ApiTokenSummary; now: number }> = ({ token, now }
...
+          minutes.
+        </p>
+        <button type="submit" class="btn">
+          Change username
+        </button>
+      </form>
+    ) : (
+      <p class="settings-help">
+        Your username is the namespace in every project and clone URL, so it cannot be changed while
+        you own projects. If you have just deleted your last project, allow up to 15 minutes.
```
[the remark this answers is not in the document]
Comment (jlamoreaux) [reply src/ui/pages/settings.tsx]: Addressed in 4fedda5. The route's check now yields `allowed`, `blocked` or `unavailable` (`UsernameChange`), and the page renders the project-ownership sentence only for `blocked`; `unavailable` still withholds the form (fail closed) but says the check could not be confirmed and asks for a reload. Both failure paths (claims unreadable, KV listing unreadable) are logged at error level. `tests/account-settings.test.tsx` asserts the retry sentence appears and the ownership sentence does not for each failure.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by jlamoreaux** [comment]
@coderabbitai review

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by jlamoreaux** [comment]
On CodeRabbit's last note (the legacy-key Rotate and Disable controls sitting in their own card rather than the Danger zone): that placement is deliberate and the PR description says the reverse of what the note assumes. The finding from the settings review was that red buttons appeared outside the Danger zone, so those two became plain buttons in a card of their own, leaving the Danger zone to the one action that destroys data (account deletion). Rotating or disabling the legacy key affects a credential, not a project or the account, and named API tokens keep working either way, so it does not belong beside "delete everything". Leaving it as is.

The risk summary's other two points are already in the diff: `getCurrentUser` forwards `displayName` to every page header (534de02), and rename vs. creation is ordered through the D1 claims table (b376db6 onward). No review thread is open and CI is green on 4fedda5.

---
_Generated by [Claude Code](https://claude.ai/code)_
